addons/auth_ldap/models/res_users.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.exceptions import AccessDenied
+
+from odoo import api, models, SUPERUSER_ID
+from odoo.modules.registry import RegistryManager
+
+
+class Users(models.Model):
+    _inherit = "res.users"
+
+    def _login(self, db, login, password):
+        user_id = super(Users, self)._login(db, login, password)
+        if user_id:
+            return user_id
+        registry = RegistryManager.get(db)
```

Comment:
instead of use of RegistryManager, can we create Enrionment ? use orm query instead of sql query.